src/Response.php

```diff
@@ -99,7 +99,7 @@ public function toResponse($request)
         $page = [
             'component' => $this->component,
             'props' => $props,
-            'url' => $request->getBaseUrl().$request->getRequestUri(),
```

Comment:
I think this can be changed to just using `fullUrl()`, https://github.com/laravel/framework/blob/10.x/src/Illuminate/Http/Request.php#L116-L123

```suggestion
            'url' => $request->fullUrl(),
```

Reply:
`fullUrl` correctly takes care of the base url and forward prefix, but it also prepends the protocol and hostname. This may have an impact on the inertia clients for the different JavaScript frameworks. It causes 10 failures in the `ResponseTest`, for example:

```
There were 10 failures:

1) Inertia\Tests\ResponseTest::test_server_response
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'/user/123'
+'http://localhost/user/123'

/ws/dev/inertia-laravel/tests/ResponseTest.php:46
/ws/dev/inertia-laravel/vendor/orchestra/testbench-core/src/PHPUnit/TestCase.php:42

....
```

The advantage of using fullUrl is that it would be more consistent, since the prop is already called `url` but in reality only contains the path URI. But it is a potentially breaking change if JS clients assume they receive only the path URI.

@RobertBoes what do you prefer?

Reply:
Hmm yeah, you're right, didn't think about that

> Also, detecting the protocol (https?) and actual client facing hostname can be a headache in a reverse proxy environment...

It shouldn't be, if the correct headers are sent and the `TrustProxies` middleware is set up properly (which is what you should do anyways if you're using a reverse proxy)